python/cudf/dataframe/dataframe.py

```diff
@@ -493,6 +493,37 @@ def __deepcopy__(self, memo={}):
             memo = {}
         return self.copy()
 
+    def __add__(self, other):
+        """
+        Parameters
+        ----------
+        other, DataFrame
+            the DataFrame to be added
+
+        Returns
+        -------
+        DataFrame
+        """
+        df = DataFrame()
+
+        matched_cols = list(set(self.columns) & set(other.columns))
+        extra_self = list(set(self.columns) - set(other.columns))
+        extra_other = list(set(other.columns) - set(self.columns))
+
+        # to maintain the order of columns in both df
+        for col in self.columns:
+            if col in matched_cols:
+                df.add_column(col, self[col] + other[col], forceindex=True)
+
+            if col in extra_self:
+                df.add_column(col, self[col], forceindex=True)
+
+        for col in other.columns:
+            if col in extra_other:
+                df.add_column(col, other[col], forceindex=True)
+
+        return df
```

Comment:
We might be able to go further and implement not just `__add__`, but many of the other binary operators like `__mul__`, `__pow__`, `__lt__`, and so on.  The logic you have here looks clean, and can probably be reused in a general function.

```python
def _apply_binary_operator(self, other, operator):
    ... your code here, but replace the `+` with the operator call

def __add__(self, other):
    return self._apply_binary_operator(other, operator)
```

I'm not sure of exactly which operators it makes sense to implement.  You might want to look at how Dask does this in `dask/utils.py::OperatorMethodMixin` and `dask/dataframe/core.py::_Frame`.  

https://github.com/dask/dask/blob/7af6fea6d3b2dde6bfaee727da57d57985db773e/dask/utils.py#L832-L867

https://github.com/dask/dask/blob/7af6fea6d3b2dde6bfaee727da57d57985db773e/dask/dataframe/core.py#L1187-L1196

https://github.com/dask/dask/blob/7af6fea6d3b2dde6bfaee727da57d57985db773e/dask/dataframe/core.py#L3231-L3255

Reply:
I have made the necessary changes to make it generic.
To support other operator, we just need to write `__sub__(), __mul__(), __pow__(), __lt__()` and call the `_apply_binary_operator(self, other, operator)` internally.